src/main/java/com/bakdata/kafka/CleanUpRunner.java

```diff
@@ -188,20 +188,24 @@ public void run(final boolean deleteOutputTopic) {
      */
     public void deleteTopics() {
         final List<String> externalTopics = this.topologyInformation.getExternalSinkTopics();
-        externalTopics.forEach(this::deleteTopic);
+        externalTopics.forEach(this::deleteTopicAndResetSchema);
     }
 
     private void resetInternalTopic(final String topic) {
-        this.adminClient.getSchemaTopicClient().resetSchemaRegistry(topic);
+        this.adminClient.getSchemaClient()
+            .ifPresent(client -> client.resetSchemaRegistry(topic));
         this.runTopicCleanUp(topic);
     }
 
     private void runTopicCleanUp(final String topic) {
         this.topicCleanUpHooks.forEach(hook -> hook.accept(topic));
     }
 
-    private void deleteTopic(final String topic) {
-        this.adminClient.getSchemaTopicClient().deleteTopicAndResetSchemaRegistry(topic);
+    private void deleteTopicAndResetSchema(final String topic) {
+        this.adminClient.getTopicClient()
```

Comment:
I think it is cleaner to leave this logic in the SchemaTopicClient. Or have two different implementations of the same interface. One SchemaTopicClient which does both and one TopicClient that only acts on topics